### DISCUSSION TITLE
New Feature: Checkbox cards

### DISCUSSION THREAD

**Discussion by RamonGal** [saadeghi/daisyui#1946] [Ideas / feature requests]
I wanted to suggest adding interactive checkbox cards that aren't joined to enhance the user experience of daisy. Currently, users can use join groups. This limits us in the way we interact with it, because checkboxes or buttons we use need to be joined.

Probably add it in the join groups with an additional class tag or something.

I am willing to contribute to the development of this feature, whether it's through feedback, coding it (fork), or any other assistance you may need.

Thank you for considering my suggestion. I believe that interactive checkbox cards would improve usability and make the selection process more intuitive. I look forward to hearing your thoughts on this idea.

**Comment by saadeghi** [comment]
If you're talking about something like [this](https://cdn.dribbble.com/users/267552/screenshots/6546524/select_reward_4x.png?compress=1&resize=1600x1200&vertical=center) or [this](https://cdn.dribbble.com/userupload/4059861/file/original-ad8bbf0cec7cb4f3761cc1b1aaf8f0cd.png?compress=1&resize=1504x1128&vertical=center), I agree. It would be a good addition. It can be a card immediately coming after a checkbox (or radio button) and selecting the card, would check the checkbox/radio.

If you mean something else, can you find a picture?

**Reply by RamonGal** [reply]
That's exactly it. 🙂
If you want, I can do it on the fork I opened. Then I'll open a pull request for review.

**Reply by saadeghi** [reply]
Sure. PRs are welcome but let's first decide how it's going to be.

Is the card going to be a big `<label>` tag and the checkbox is going to sit immediately before it?

In that case we can't have other form elements inside the card.

And If the checkbox is inside the card, we can't style the parent card (not until CSS `has()` reaches a good browser support)

And for the card style, I think an outline similar to button's `focus-visible` outline would be good.

**Reply by RamonGal** [reply]
I can think of 3 methods:

- Leave it to javascript, take care of the classes :
```vue
<template>
  <div class="w-full px-4 py-4">
    <div class="mx-auto w-full max-w-md">
      <div class="box-group">
        <template v-for="item in boxGroup" :key="item.id">
          <label
            tabindex="0"
            class="box-card m-1"
            :class="selected[item.id] ? 'box-card-active' : ''"
            @click="selected[item.id] = !selected[item.id]"
          >
            <div class="flex w-full items-center justify-between">
              <div class="flex items-center">
                <div class="text-sm">
                  <p class="font-medium">{{ item.title }}</p>
                  <p>{{ item.context }}</p>
                </div>
              </div>
              <div
                class="box-card-simbol"
                :class="selected[item.id] ? 'box-card-simbol-active' : ''"
              >
                <CheckIcon class="w-5 h-5" />
              </div>
            </div>
          </label>
        </template>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { ref } from 'vue';
import { CheckIcon } from '@heroicons/vue/24/solid';

const boxGroup = [
  {
    id: 0,
    title: 'Premium',
    context:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer posuere erat a ante.'
  },
  {
    id: 1,
    title: 'Basic',
    context:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer posuere erat a ante.'
  },
  {
    id: 2,
    title: 'Standard',
    context:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer posuere erat a ante.'
  }
];

const selected = ref([false, false, false]);
</script>

<style>
.box-card {
  @apply bg-white relative flex cursor-pointer rounded-lg px-5 py-4 shadow-md focus:outline-0;

  &-active {
    @apply bg-primary bg-opacity-75 text-white relative flex cursor-pointer rounded-lg shadow-md focus:outline-none focus:ring focus:ring-primary;
  }
}

.box-card-simbol {
  @apply bg-white text-white relative flex cursor-pointer p-3 rounded-full shadow-lg focus:outline-none;
  &-active {
    @apply text-white bg-neutral p-3 rounded-full;
  }
}
</style>

```
Just an example, can change to use the _bg_ or _focus_ like you said and add an invisible checkbox binded to the value
 
![Screenshot 2023-06-08 002753](https://github.com/saadeghi/daisyui/assets/38587709/acefe67a-b357-462d-a90a-830fe1369cb4)


In this case we can bind the value of the checkbox to the string.

- Pseudo classes, for checking inner  : 
```css
div input[type="checkbox"].box-checkbox:checked {
  background-color: #ccc;
}
```

- Using the labels like you said, but if you add any form data in it, clicking them would select the checkbox.

**Reply by RamonGal** [reply]
are there any limitations on the way the components are to be in Daisy? e.g.: being pure css and html
